src/app/Models/Post.php

```diff
@@ -54,4 +55,9 @@ public function originalPost(): BelongsTo
     {
         return $this->belongsTo(Post::class, 'original_post_id');
     }
+
+    public function getUserPostAttribute(): Bool
+    {
+        return Auth::check() && optional($this->user)->id === Auth::id();
+    }
```

Comment:
You're getting user posts, but it appears that the method is returning a conditional attribute. Furthermore, the name of the attribute does not accurately reflect what the method returns.